### ISSUE TITLE
ZuluFX Windows desktop build app name

### ISSUE THREAD

**Issue by jsfan3** [codenameone/CodenameOne#3266]
I am working on a desktop build for Windows. I noticed that if I use ZuluFX as described in https://www.codenameone.com/blog/zulu-desktop-builds.html, the name of the application the user will see is the same as the main class name instead of the app name. I am referring specifically to the link created on the desktop, I assume it is a small bug.

I also noticed that a link is missing in the list of all applications.

**Comment by jsfan3** [comment]
The same issue is for the installation directory: the installer of the ZuluFX version of the app defaults to a path using the main class name instead of the app name.

**Comment by jsfan3** [comment]
@shai-almog Windows Desktop Build is a Pro feature, but I'm currently on the free plan, so I can't tell you if the issue still exists.

**Comment by shai-almog** [comment]
@jsfan3 upgraded you.

**Comment by jsfan3** [comment]
Thank you so much for the upgrade, Shai!

Unfortunately, **the issue has been confirmed**, but it's not related to ZuluFX.
I'm currently using Eclipse on Linux. The build log reports:
Java version: 23.0.1, vendor: Eclipse Adoptium, runtime: /home/francesco/.p2/pool/plugins/

You can note that:
- the installation directory defaults to the main class name rather than the application name.
- the name of the application links in the desktop and "All Programs" is the same as the main class name, rather than the application name.

Also, the built exe contains the main class name instead of the app name, but this isn't very important.

**Comment by shai-almog** [comment]
I'm afraid these are things we probably can't fix as this will break compatibility and won't allow us to update existing installs. I think this was done intentionally though, class name would typically be in latin script which is guaranteed to work for app of the above on all versions of windows. Application name can be localized which might be problematic for some scripts.

**Comment by jsfan3** [comment]
Yes, it makes sense that changing this behavior could affect compatibility with updates. In particular, it makes sense if we assume that the user leaves the installation folder as suggested by the installer.

However, having the application shortcuts (on the desktop and in “All Programs”) named “MainClass” or something similar is not desirable.

One solution to maintain compatibility while allowing customization of the installer is to add two build hints: one to specify the default installation folder, and the other to specify the default link names. These build hints could be managed automatically by the “Control Center.” Currently, “Device Settings” only includes options for “Windows UWP builds.” Here you could add a tab for options related to “Windows desktop builds.” This would make everything clearer and more customizable.

What do you think?

**Comment by shai-almog** [comment]
That makes sense. I'll take another stab at this.

**Comment by jsfan3** [comment]
@shai-almog Thank you very much!
